### PR TITLE
Remove deepcopy from UMFPACK, add kwarg to copy

### PR DESCRIPTION
**Pull request by Wimmerer**
@JeffBezanson is this more acceptable? The semantics of the previous version didn't really match `deepcopy` anyway (only certain fields were copied). 

Can `copy` signature be modified without affecting Base specialization changes?

### CODE REVIEW

**Comment by Wimmerer** [comment]
@SobhanMP is this an acceptable interface for copying? I believe you were the primary person looking for it (for multicore stuff?). I've made a PR to SuiteSparse upstream to add copying, it should hopefully be merged before feature freeze.

**Comment by SobhanMP** [comment]
looks fine to me.

**Comment by SobhanMP** [comment]
might want to also deal with 

https://github.com/JuliaSparse/SparseArrays.jl/blob/e15a37e5ec722aabb66229d82890ccd0e32182a7/src/solvers/umfpack.jl#L282

**Comment by Wimmerer** [comment]
I don't think this needs a backport @KristofferC SuiteSparse isn't in this repo in 1.8

**Comment by JeffBezanson** [comment]
Thanks, this is better. Is `safecopy` so expensive that it can't be the default? Shame to need that option.

**Comment by SobhanMP** [comment]
That's almost all of the computation when calculating the LU factorization. Since we rename duplicate to copy, copy should be as cheap as possible for multi-threaded applications.

**Comment by Wimmerer** [comment]
There may be an improved default (implemented as a memcpy) once SuiteSparse upstream can be updated (I have a PR in, it just needs some tests and rebasing). That avoids the recomputation of the LU, which is what has to be done now for safecopy. I'm hoping to enable this for all the SuiteSparse solvers, but I don't know Tim's SuiteSparse release cycle timeline.

TLDR: Default can be made safe and fast ~Julia 1.10
